.github/workflows/build.yml

```diff
@@ -114,7 +114,9 @@ jobs:
       - uses: ./.github/actions/setup
       - name: build test container
         run: |
-          podman build --squash --tag test --build-arg base="$(./build --print-container-image)" tests
+          OCI_IMAGE_TAG="$(./build --print-container-image)"
+          podman build -t "$OCI_IMAGE_TAG" -f builder.dockerfile .
```

Comment:
this now assumes the existence of `builder.dockerfile`, which I think is okay in this case. It would be possible to make it more generic, but since this is not re-usable library code here, I think that's okay. Simplicity over being generic.